--- awk.h.orig	2026-07-09 23:47:00.000000000 +0700
+++ awk.h	2026-09-09 08:16:13.870996287 +0700
@@ -175,7 +175,21 @@
 #define setlocale(locale, val)	/* nothing */
 #endif /* HAVE_SETLOCALE */
 
-#ifdef HAVE_FWRITE_UNLOCKED
+#if defined(__ANDROID__)
+static inline size_t android_fwrite_unlocked(const void *ptr, size_t size, size_t nitems, FILE *stream) {
+    const char *cptr = ptr;
+    size_t cnt = size * nitems;
+    while (cnt && EOF != putc_unlocked(*cptr, stream)) {
+        cptr++;
+        cnt--;
+    }
+    return (cptr - (const char*)ptr) / size;
+}
+#define fwrite_unlocked android_fwrite_unlocked
+#define HAVE_FWRITE_UNLOCKED 1
+#endif
+
+#if defined(HAVE_FWRITE_UNLOCKED) && !defined(__ANDROID__)
 #define fwrite	fwrite_unlocked
 #endif /* HAVE_FWRITE_UNLOCKED */
 
