diff -u -r /data/data/com.termux/files/home/gawk-5.4.1/builtin.c /data/data/com.termux/files/home/gawk-5.4.1-mod/builtin.c
--- /data/data/com.termux/files/home/gawk-5.4.1/builtin.c	2026-07-09 23:47:00.000000000 +0700
+++ /data/data/com.termux/files/home/gawk-5.4.1-mod/builtin.c	2026-09-09 14:33:50.596386018 +0700
@@ -3421,7 +3421,7 @@
 		istat = signal(SIGINT, istat);
 		qstat = signal(SIGQUIT, qstat);
 #endif
-		execl("/bin/sh", "sh", "-c", command, NULL);
+		execl("/data/data/com.termux/files/usr/bin/sh", "sh", "-c", command, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 
 		return 0;	// silence compiler warnings
diff -u -r /data/data/com.termux/files/home/gawk-5.4.1/io.c /data/data/com.termux/files/home/gawk-5.4.1-mod/io.c
--- /data/data/com.termux/files/home/gawk-5.4.1/io.c	2026-07-09 23:47:00.000000000 +0700
+++ /data/data/com.termux/files/home/gawk-5.4.1-mod/io.c	2026-09-09 14:33:50.592386018 +0700
@@ -2080,7 +2080,7 @@
 
 		set_sigpipe_to_default();
 
-		execl("/bin/sh", "sh", "-c", command, NULL);
+		execl("/data/data/com.termux/files/usr/bin/sh", "sh", "-c", command, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 
 	case -1:
@@ -2132,7 +2132,7 @@
 
 		signal(SIGPIPE, SIG_DFL);
 
-		execl("/bin/sh", "sh", "-c", command, NULL);
+		execl("/data/data/com.termux/files/usr/bin/sh", "sh", "-c", command, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 
 	case -1:
@@ -2471,7 +2471,7 @@
 			fatal(_("close of pipe failed: %s"), strerror(errno));
 		/* stderr does NOT get dup'ed onto child's stdout */
 		set_sigpipe_to_default();
-		execl("/bin/sh", "sh", "-c", str, NULL);
+		execl("/data/data/com.termux/files/usr/bin/sh", "sh", "-c", str, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 	}
 #endif /* NOT __MINGW32__ */
@@ -2719,7 +2719,7 @@
 		if (close(p[0]) == -1 || close(p[1]) == -1)
 			fatal(_("close of pipe failed: %s"), strerror(errno));
 		set_sigpipe_to_default();
-		execl("/bin/sh", "sh", "-c", cmd, NULL);
+		execl("/data/data/com.termux/files/usr/bin/sh", "sh", "-c", cmd, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 	}
 #endif /* NOT __MINGW32__ */
@@ -4731,7 +4731,7 @@
 			fatal(_("gawk_popen_write: failed to move pipe fd to standard input"));
 		(void) close(pipefds[0]);
 		set_sigpipe_to_default();
-		execl("/bin/sh", "sh", "-c", cmd, NULL);
+		execl("/data/data/com.termux/files/usr/bin/sh", "sh", "-c", cmd, NULL);
 		_exit(errno == ENOENT ? 127 : 126);
 	} else if (childpid < 0) {
 		(void) close(pipefds[0]);
